src/cpp/fastdds/xtypes/type_conversion/dyn_type_tree.ipp

```diff
@@ -0,0 +1,1116 @@
+        std::string& type_str) noexcept
+{
+    ReturnCode_t ret = RETCODE_OK;
+
+    switch (dyn_type->get_kind())
+    {
+        case TK_BOOLEAN:
+        {
+            type_str = "boolean";
+            break;
+        }
+        case TK_BYTE:
+        {
+            type_str = "octet";
+            break;
+        }
+        case TK_INT8:
+        {
+            type_str = "char";
+            break;
+        }
+        case TK_INT16:
+        {
+            type_str = "short";
+            break;
+        }
+        case TK_INT32:
+        {
+            type_str = "long";
+            break;
+        }
+        case TK_INT64:
+        {
+            type_str = "long long";
+            break;
+        }
+        case TK_UINT8:
+        {
+            type_str = "unsigned char";
+            break;
+        }
+        case TK_UINT16:
+        {
+            type_str = "unsigned short";
+            break;
+        }
+        case TK_UINT32:
+        {
+            type_str = "unsigned long";
+            break;
+        }
+        case TK_UINT64:
+        {
+            type_str = "unsigned long long";
+            break;
+        }
+        case TK_FLOAT32:
+        {
+            type_str = "float";
+            break;
+        }
+        case TK_FLOAT64:
+        {
+            type_str = "double";
+            break;
+        }
+        case TK_FLOAT128:
+        {
+            type_str = "long double";
+            break;
+        }
+        case TK_CHAR8:
+        {
+            type_str = "char";
+            break;
+        }
+        case TK_CHAR16:
+        {
+            type_str = "wchar";
+            break;
+        }
+        case TK_STRING8:
+        {
+            type_str = "string";
+            break;
+        }
+        case TK_STRING16:
+        {
+            type_str = "wstring";
+            break;
+        }
+        case TK_ARRAY:
+        {
+            ret = array_kind_to_str(dyn_type, type_str);
+            break;
+        }
+        case TK_SEQUENCE:
+        {
+            ret = sequence_kind_to_str(dyn_type, type_str);
+            break;
+        }
+        case TK_MAP:
+        {
+            ret = map_kind_to_str(dyn_type, type_str);
+            break;
+        }
+        case TK_ALIAS:
+        case TK_BITMASK:
+        case TK_BITSET:
+        case TK_ENUM:
+        case TK_STRUCTURE:
+        case TK_UNION:
+        {
+            type_str = dyn_type->get_name().to_string();
+            break;
+        }
+        case TK_NONE:
+        {
+            ret = RETCODE_UNSUPPORTED;
+            break;
+        }
+        default:
+        {
+            ret = RETCODE_BAD_PARAMETER;
+            break;
+        }
+    }
+
+    return ret;
+}
+
+ReturnCode_t array_kind_to_str(
+        const DynamicType::_ref_type& dyn_type,
+        std::string& array_str) noexcept
+{
+    if (dyn_type->get_kind() != TK_ARRAY)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not an array.");
```

Comment:
I've been using warnings in the DynamicData->JSON part, not sure which one is better, but I guess we should be using the same kind.

Reply:
I believe errors are more pertinent.

Reply:
Ok, changed warnings for errors on my part.